azurerm/internal/services/resource/tests/resource_arm_provider_registration_test.go

```diff
@@ -0,0 +1,99 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestAccAzureRMResourceProviderRegistration_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_resource_provider_registration", "test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acceptance.PreCheck(t) },
+		Providers:    acceptance.SupportedProviders,
+		CheckDestroy: testCheckAzureRMResourceProviderDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMResourceProviderRegistration_basic(),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMResourceProviderRegistrationExists(data.ResourceName),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "resource_provider_namespace"),
+				),
+			},
```

Comment:
can we also add an import check here to check this imports correctly:

```suggestion
			},
			data.ImportStep(),
```